hypertrace-alert-engine/src/integrationTest/java/org/hypertrace/alert/engine/HypertraceAlertEngineTest.java

```diff
@@ -165,43 +167,95 @@ public static void setup() throws Exception {
     LOG.info("TraceTimeStamp for trace is {}", Instant.ofEpochMilli(traceTimeStamp));
     assertTrue(generateData(traceTimeStamp));
     LOG.info("Generate Data Complete");
-
-    mockWebServer = new MockWebServer();
-    mockWebServer.start(NOTIFICATION_CHANNEL_PORT);
-    MockResponse mockedResponse =
-        new MockResponse().setResponseCode(200).addHeader("Content-Type", "application/json");
-    mockWebServer.enqueue(mockedResponse);
-    withEnvironmentVariable("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
-        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
-        .and("QUERY_SERVICE_HOST_CONFIG", queryService.getHost())
-        .and("QUERY_SERVICE_PORT_CONFIG", queryService.getMappedPort(8090).toString())
-        .execute(
-            () ->
-                IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"}));
-    Assertions.assertEquals(0, mockWebServer.getRequestCount());
   }
 
   @AfterAll
-  public static void shutdown() throws IOException {
+  public static void shutdown() {
     LOG.info("Initiating shutdown");
     attributeService.stop();
     queryService.stop();
     mongo.stop();
     pinotServiceManager.stop();
     kafkaZk.stop();
     network.close();
-    mockWebServer.close();
   }
 
   @Test
-  public void testNotificationSent() throws InterruptedException {
+  public void testStaticThresholdNotificationSent() throws Exception {
+    LOG.info("Starting testStaticThresholdNotificationSent");
+    MockWebServer mockWebServer = new MockWebServer();
+    mockWebServer.start(NOTIFICATION_CHANNEL_PORT1);
+    MockResponse mockedResponse =
+        new MockResponse().setResponseCode(200).addHeader("Content-Type", "application/json");
+    mockWebServer.enqueue(mockedResponse);
+
+    Assertions.assertEquals(0, mockWebServer.getRequestCount());
+    withEnvironmentVariable("ATTRIBUTE_SERVICE_HOST_CONFIG", attributeService.getHost())
+        .and("ATTRIBUTE_SERVICE_PORT_CONFIG", attributeService.getMappedPort(9012).toString())
+        .and("QUERY_SERVICE_HOST_CONFIG", queryService.getHost())
+        .and("QUERY_SERVICE_PORT_CONFIG", queryService.getMappedPort(8090).toString())
+        .and("ALERT_RULES_PATH", "build/resources/integrationTest/alert-rules-static-rule.json")
+        .and(
+            "NOTIFICATION_CHANNELS_PATH",
+            "build/resources/integrationTest/notification-channels-1.json")
+        .and("SERVICE_ADMIN_PORT", "10005")
+        .and("JOB_SUFFIX", "job1")
+        .execute(
+            () -> {
+              ConfigFactory.invalidateCaches();
+              IntegrationTestServerUtil.startServices(new String[] {"hypertrace-alert-engine"});
+            });
     // int x = 1;
     int retryCount = 0;
-    while (!(mockWebServer.getRequestCount() > 0) && retryCount++ < 10) {
+    while (!(mockWebServer.getRequestCount() == 1) && retryCount++ < 10) {
```

Comment:
`== 1` is tight condition, I think, you can keep `> 0` which is serving the purspose.